core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +376,174 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
     }
   }
 
-  private List<String> toRoleNames(List<RoleEntity> roleEntities) {
-    return roleEntities.stream().map(RoleEntity::name).collect(Collectors.toList());
+  Role grantPrivilegeToRole(
+      String metalake, String role, MetadataObject object, List<Privilege> privileges) {
+    try {
+      return store.update(
+          AuthorizationUtils.ofRole(metalake, role),
+          RoleEntity.class,
+          Entity.EntityType.ROLE,
+          roleEntity -> {
+            List<SecurableObject> securableObjects = roleEntity.securableObjects();
+
+            boolean objectExist = false;
+
+            List<SecurableObject> updateSecurableObjects = Lists.newArrayList();
+            // Only update the first matched securable object privilege
+            for (SecurableObject securableObject : securableObjects) {
```

Comment:
Is it possible that there are many securable objects that equals to the `object` in the List?

Reply:
Yes, but it's enough that we add the first matched securable object like the comment.

Reply:
Why not remove duplicated items in the `roleEntity`?

Reply:
OK, I can add the limitation for them.